sample/ml_sample_validator.json

```diff
@@ -0,0 +1,93 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://raw.githubusercontent.com/EMBL-EBI-SUBS/validation-schemas/master/sample/ml_sample_validator.json",
+  "title": "MetaboLights Sample Validator Schema",
+  "description": "Schema based on rules found in http://isa-specs.readthedocs.io/en/latest/isajson.html#sample-schema-json",
+  "version": "1.0.0",
+  "author": "metabolights",
+  "type": "object",
+  "properties": {
+
+    "alias": {
+      "description": "A sample unique identifier in a submission.",
+      "type": "string"
+    },
+    "taxonId": {
+      "description": "The taxonomy id for the sample species.",
+      "type": "integer"
+    },
+    "taxon": {
+      "description": "The taxonomy name for the sample species.",
+      "type": "string"
+    },
+    "title": {
+      "description": "A sample title.",
+      "type": "string"
+    },
+    "releaseDate": {
+      "description": "Date from which this sample is released publicly.",
+      "type": "string",
+      "format": "date"
+    },
+    "description": {
+      "description": "A brief description of the sample including species name.",
+      "type": "string"
+    },
+    "sampleRelationships": {
+      "description": "List of relationships between samples.",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "target": {
+            "description": "Sample accession to which the sample relationship refers to.",
+            "type": "string",
+            "pattern": "^SAM[END][AG]?[0-9]+"
+          },
+          "nature": {
+            "description": "Nature of the relationship between the samples.",
+            "type": "string",
+            "enum": ["derived from", "same as", "child of"]
+          }
+        },
+        "required": ["target", "nature"]
+      }
+    }
+  },
+  "attributes": {
+      "description": "Attributes for describing a sample.",
+      "type": "object",
+      "properties": {
+        "Organism": {
+          "description": "Sample organism source"
+        },
+        "Organism part": {
+          "description": "Sample organism part source"
+          }
+        },
+        "required": ["Organism","Organism part"]
+      },
+    "patternProperties": {
+      "^.*$": {
+        "type": "array",
+        "minItems": 1,
+        "items": {
+          "properties": {
+            "value": { "type": "string", "minLength": 1 },
+            "units": { "type": "string", "minLength": 1 },
+            "terms": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "url": {"type": "string", "format": "uri" }
+                },
+                "required": ["url"]
+              }
+            }
+          },
+          "required": ["value"]
+        }
+      }
+    }
+}
```

Comment:
There's a missing _required_ on the root object and since there are two required _attributes_, for that requirement to be imposed, the attributes must also be required in the root object, so that the required section will look like: 
```js
"required": [ "alias", "taxonId", "releaseDate", "attributes" ]
```

Reply:
Oops, added!